keras/src/layers/rnn/rnn.py

```diff
@@ -428,7 +428,7 @@ def call(
             output = last_output
 
         if self.return_state:
-            if len(states) == 1:
+            if len(ops.shape(states)) == 1:
```

Comment:
State is always intended to be a list here, because the last element returned by `self.inner_loop` (states) should be a list.